### DISCUSSION TITLE
Does AvaloniaUI have an icon based toolbar component?

### DISCUSSION THREAD

**Discussion by futuremotiondev** [AvaloniaUI/Avalonia#10939] [General]
Example: 
[https://help.syncfusion.com/wpf/toolbar/getting-started](https://help.syncfusion.com/wpf/toolbar/getting-started)
or 
[https://wpf-tutorial.com/common-interface-controls/toolbar-control/](https://wpf-tutorial.com/common-interface-controls/toolbar-control/)
(This is old and looks bad but conveys what I am looking for)

Some images for reference:

![syncfusion samplebrowser wpf_NoTNP2MsSa](https://user-images.githubusercontent.com/33441569/230516064-6710d767-93e6-4c8d-9f29-5575ba4866b7.png)

![syncfusion samplebrowser wpf_JMGOH7Qew9](https://user-images.githubusercontent.com/33441569/230516071-31cebde2-275d-4196-ba41-851718ab5af8.png)

I think I can do this with a horizontal stack panel, but it would be nice to have an existing component. Also kind of confused how to add SVGs to my toolbar if I go this route.

Does anyone know if Avalonia has something like this? I don't see it in the [docs](https://docs.avaloniaui.net/docs/controls).

Thanks for any help.

**Comment by futuremotiondev** [comment]
So with no answers available I decided to roll my own using a combination of `Menu`, `DockPanel`, `Grid`, and `Border` plus buttons and other decorations:

![AvaloniaUITest1_Y7Ml8LE0Ws](https://user-images.githubusercontent.com/33441569/230547112-9db5b7e9-bb83-4abf-8b39-4a66f2566342.png)

There is no business logic or anything really in terms of code. Just a skeleton front-end. But in case it might help someone working on an application that needs a toolbar, I've uploaded the entire project to this post.

Note: I deleted the bin folder because it was over 100mb. You'll have to build the project once before the AXAML preview will work.

Hope this might help someone in the future.

[AvaloniaUITest1.zip](https://github.com/AvaloniaUI/Avalonia/files/11176295/AvaloniaUITest1.zip)

**Comment by mysteryx93** [comment]
One downside of Avalonia is limited 3rd party support for more advanced UI. Hopefully that will change in the future.

**Comment by andrewdbond** [comment]
Does something like AvaloniaRibbon help? https://github.com/AvaloniaCommunity/awesome-avalonia#controls

**Reply by timunie** [reply]
FluentAvalonia has also a kind of tool bar available. Nevertheless we hope to get some paid controls like telerik and co on board for companies requring instant support. Small OSS teams can't handle this. 🤞
